### PR TITLE
fix(hardon-document): cancel edit on non-existent field COMPASS-6505

### PR DESCRIPTION
**Pull request by mabaasit**
When cancelling from editing a field in a document that's non-existent, the compass throws an error.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by addaleax — APPROVED** [review]
Should this have a test?

**Review by addaleax — APPROVED** [review]

File: packages/hadron-document/src/element.ts [hunk 1]
```diff
@@ -699,7 +699,7 @@ export class Element extends EventEmitter {
    */
   revert(): void {
     if (this.isAdded()) {
-      this.parent!.elements!.remove(this);
+      this.parent?.elements?.remove(this);
       this._bubbleUp(Events.Removed, this, this.parent);
```
Comment (mcasimir) [inline packages/hadron-document/src/element.ts:703]: is it ok that `this._bubbleUp` is triggered anyway?

File: packages/hadron-document/src/element.ts [hunk 2]
```diff
@@ -699,7 +699,7 @@ export class Element extends EventEmitter {
    */
   revert(): void {
     if (this.isAdded()) {
-      this.parent!.elements!.remove(this);
+      this.parent?.elements?.remove(this);
```
Comment (mcasimir) [inline packages/hadron-document/src/element.ts:702]: just a question, in which case do we end up with a document that doesn't exist?

**Review by mcasimir — APPROVED** [review]
🔥

File: packages/hadron-document/src/element.ts (hunk 2 above)
Comment (mabaasit) [reply packages/hadron-document/src/element.ts:702]: i tried really hard to find the root cause, but was not. I am giving up and will merge this PR.

**Pull request merged**